### ISSUE TITLE
Setting DialogHost.IsOpen to false does not trigger DialogClosing event

### ISSUE THREAD

**Issue by pfeigl** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1181]
The DialogClosing event is emitted in the `Close` Method, however this method is never called, when the dialog is closed via the IsOpen property.

Thus the `DialogClosing` event is never emitted.

Is this expected behavior or a bug?

**Comment by Keboo** [comment]
@pfeigl somewhat expected. The `DialogClosing` event [allows for cancelation](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MaterialDesignThemes.Wpf/DialogClosingEventArgs.cs#L35-L43). If it were raised when `IsOpen` change that could then mean that when you set the property to false that if you immediately read it again, it would still be false. 
```C#
DialogHost dialogHost = ...;
dialogHost.IsOpen = false;
bool isOpen = dialogHost.IsOpen;
//isOpen could be true at this point if closing was cancelled.
```

With that said, I do agree that it is a bit confusing. The DialogHost is one area where I think the API could use some work. I am open to suggestions on how it should work.

**Comment by pfeigl** [comment]
I need to dig a little more into the code to give some good feedback. Some things that come into my mind:

* Have an `IsCancelable` property on the closing event and throw if a non cancelable event is beeing canceled
* Keep the ClosingEvent as is, put meaningful documentation (comment) into the code but additionally emit an DialogClosed event which is always thrown. Except from stopping the closing, for most situations it's probably irrelevant whether people register on Closing or Closed, they probably just want to do something after the dialog is "gone".

**Comment by Keboo** [comment]
I think adding in a new event for `DialogClosed` makes the most sense here. This event should always be raised.

**Comment by Keboo** [comment]
Resolved by #2767
